File: app/src/main/java/com/papi/nova/nvstream/ConnectionContext.java
package com.papi.nova.nvstream;

import com.papi.nova.nvstream.http.ComputerDetails;

import java.security.cert.X509Certificate;

import javax.crypto.SecretKey;

public class ConnectionContext {
    public ComputerDetails.AddressTuple serverAddress;
    public int httpsPort;
    public boolean isNvidiaServerSoftware;
    public X509Certificate serverCert;
    public StreamConfiguration streamConfig;
    public NvConnectionListener connListener;
    public SecretKey riKey;
    public int riKeyId;
    
    // This is the version quad from the appversion tag of /serverinfo
    public String serverAppVersion;
    public String serverGfeVersion;
    public int serverCodecModeSupport;
    public int serverMaxLaunchRefreshRate;

    // This is the sessionUrl0 tag from /resume and /launch
    public String rtspSessionUrl;
    public String sessionToken;
    public Boolean currentGameOwnedByClient;
    public String currentGameOwnerName;
    public boolean watchOnlyRequested;
    
    public int negotiatedWidth, negotiatedHeight;
    public boolean negotiatedHdr;
    public float negotiatedLaunchRefreshRate;

    public int negotiatedRemoteStreaming;
    public int negotiatedPacketSize;

    public int videoCapabilities;
}

File: app/src/main/java/com/papi/nova/nvstream/ConnectionContext.kt
package com.papi.nova.nvstream

import com.papi.nova.nvstream.http.ComputerDetails
import java.security.cert.X509Certificate
import javax.crypto.SecretKey

class ConnectionContext {
    @JvmField var serverAddress: ComputerDetails.AddressTuple? = null
    @JvmField var httpsPort = 0
    @JvmField var isNvidiaServerSoftware = false
    @JvmField var serverCert: X509Certificate? = null
    @JvmField var streamConfig: StreamConfiguration? = null
    @JvmField var connListener: NvConnectionListener? = null
    @JvmField var riKey: SecretKey? = null
    @JvmField var riKeyId = 0

    @JvmField var serverAppVersion: String? = null
    @JvmField var serverGfeVersion: String? = null
    @JvmField var serverCodecModeSupport = 0
    @JvmField var serverMaxLaunchRefreshRate = 0

    @JvmField var rtspSessionUrl: String? = null
    @JvmField var sessionToken: String? = null
    @JvmField var currentGameOwnedByClient: Boolean? = null
    @JvmField var currentGameOwnerName: String? = null
    @JvmField var watchOnlyRequested = false
    // What this device offers before a watch narrows the handshake to one stream's codec and
    // depth. A second refusal for another mode is checked against this, never the narrowed list.
    @JvmField var deviceVideoFormats: Int? = null
    @JvmField var decodesHevcTenBit = false
    @JvmField var decodesAv1TenBit = false
    @JvmField var hdrRequestedInSettings = false

    @JvmField var negotiatedWidth = 0
    @JvmField var negotiatedHeight = 0
    @JvmField var negotiatedHdr = false
    @JvmField var negotiatedLaunchRefreshRate = 0f

    @JvmField var negotiatedRemoteStreaming = 0
    @JvmField var negotiatedPacketSize = 0
    @JvmField var videoCapabilities = 0
}
